### PR TITLE
LibWeb: Add CanvasRenderingContext2D spec links

### PR DESCRIPTION
**Pull request by jdahlin**
Add spec links and FIXMEs

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/CanvasRenderingContext2D.cpp [hunk 1]
```diff
@@ -58,38 +58,59 @@ JS::NonnullGCPtr<HTMLCanvasElement> CanvasRenderingContext2D::canvas_for_binding
     return *m_element;
 }
 
+// 4.12.5.1.10 https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-fillrect
```
Comment (linusg) [inline Userland/Libraries/LibWeb/HTML/CanvasRenderingContext2D.cpp:61]: Please omit the section number, which is for https://html.spec.whatwg.org/multipage/canvas.html#drawing-rectangles-to-the-bitmap

File: Userland/Libraries/LibWeb/HTML/CanvasRenderingContext2D.cpp [hunk 2]
```diff
@@ -58,38 +58,59 @@ JS::NonnullGCPtr<HTMLCanvasElement> CanvasRenderingContext2D::canvas_for_binding
...
+
     auto& drawing_state = this->drawing_state();
 
+    // 3. Paint the specified rectangular area using this's fill style.
     auto rect = drawing_state.transform.map(Gfx::FloatRect(x, y, width, height));
     painter->fill_rect(enclosing_int_rect(rect), drawing_state.fill_style);
     did_draw(rect);
 }
 
+// 4.12.5.1.10 https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-clearrect
```
Comment (linusg) [inline Userland/Libraries/LibWeb/HTML/CanvasRenderingContext2D.cpp:80]: Likewise

File: Userland/Libraries/LibWeb/HTML/CanvasRenderingContext2D.cpp [hunk 3]
```diff
@@ -58,38 +58,59 @@ JS::NonnullGCPtr<HTMLCanvasElement> CanvasRenderingContext2D::canvas_for_binding
     return *m_element;
 }
 
+// 4https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-fillrect
```
Comment (davidot) [inline Userland/Libraries/LibWeb/HTML/CanvasRenderingContext2D.cpp:61]: I think you just missed this `4` in front of the link here

Comment (davidot) [inline Userland/Libraries/LibWeb/HTML/CanvasRenderingContext2D.cpp:61]: Looks like you missed the `4` just in front of the link here

File: Userland/Libraries/LibWeb/HTML/CanvasRenderingContext2D.cpp [hunk 4]
```diff
@@ -58,38 +58,59 @@ JS::NonnullGCPtr<HTMLCanvasElement> CanvasRenderingContext2D::canvas_for_binding
...
 
+// https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-strokerect
 void CanvasRenderingContext2D::stroke_rect(float x, float y, float width, float height)
 {
     auto painter = this->painter();
     if (!painter)
         return;
 
+    // FIXME: 1. If any of the arguments are infinite or NaN, then return.
+
```
Comment (davidot) [inline Userland/Libraries/LibWeb/HTML/CanvasRenderingContext2D.cpp:105]: And here is some more trailing white space

**Review by davidot — CHANGES_REQUESTED** [review]
Thanks for removing the numbers, just found one last one.
Also the clang-format is complaining about some white space

And also could you squash the commits? No need to make another PR, you can try to follow this video: https://youtu.be/ElRzTuYln0M or ask for help on the discord or here.

**Pull request closed without merging**